api/parts/data/cache.js

```diff
@@ -721,38 +721,31 @@ class CacheMaster {
  */
 function createCollection(db, name, size = 1e7) {
     return new Promise((resolve, reject) => {
-        db.onOpened(() => {
-            if (!db._native) {
-                return setTimeout(() => {
-                    createCollection(db, name, size).then(resolve, reject);
-                }, 1000);
+        db.createCollection(name, {capped: true, size: size}, (e, col) => {
```

Comment:
All MongoDB driver functions return a promise if they do not receive a callback, so you may want to give a try at going full async/await, no ?

Reply:
This would give an ability to write new code with async/await code yes
rewriting all the old code is a completely different thing :D